Git-commit: e14930e9f9c657cf109326752871d3c701b12326
Message-Id: <e14930e9f9c657cf109326752871d3c701b12326.1657695708.git.geliang.tang@suse.com>
In-Reply-To: <6c32496964da0dc230cea763a0e934b2e02dabd5.1657695708.git.geliang.tang@suse.com>
References: <6c32496964da0dc230cea763a0e934b2e02dabd5.1657695708.git.geliang.tang@suse.com>
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 7 Jul 2022 16:07:53 -0700
Subject: [PATCH] xfs: remove incorrect ASSERT in xfs_rename
References: bsn#19
Patch-mainline: v5.10.130

commit e445976537ad139162980bee015b7364e5b64fff upstream.

This ASSERT in xfs_rename is a) incorrect, because
(RENAME_WHITEOUT|RENAME_NOREPLACE) is a valid combination, and
b) unnecessary, because actual invalid flag combinations are already
handled at the vfs level in do_renameat2() before we get called.
So, remove it.

Reported-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
Reviewed-by: Darrick J. Wong <djwong@kernel.org>
Signed-off-by: Darrick J. Wong <djwong@kernel.org>
Fixes: 7dcf5c3e4527 ("xfs: add RENAME_WHITEOUT support")
Signed-off-by: Kuniyuki Iwashima <kuniyu@amazon.com>
Acked-by: Darrick J. Wong <djwong@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 fs/xfs/xfs_inode.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/xfs/xfs_inode.c b/fs/xfs/xfs_inode.c
index e958b1c74561..03497741aef7 100644
--- a/fs/xfs/xfs_inode.c
+++ b/fs/xfs/xfs_inode.c
@@ -3170,7 +3170,6 @@ xfs_rename(
 	 * appropriately.
 	 */
 	if (flags & RENAME_WHITEOUT) {
-		ASSERT(!(flags & (RENAME_NOREPLACE | RENAME_EXCHANGE)));
 		error = xfs_rename_alloc_whiteout(target_dp, &wip);
 		if (error)
 			return error;
-- 
2.34.1

